connection.go

```diff
@@ -86,12 +88,41 @@ func newTCPConnection(address string, timeout, readTimeout time.Duration) (*conn
 		respc:       make(map[int32]chan []byte),
 		logger:      &nullLogger{},
 		readTimeout: readTimeout,
+		apiVersions: make(map[int16]proto.SupportedVersion),
 	}
 	go c.nextIDLoop()
 	go c.readRespLoop()
 	return c, nil
 }
 
+//getBestVersion returns version for passed apiKey which best fit server and client requirements
+func (c *connection) getBestVersion(apiKey int16) int16 {
+	if requested, ok := c.apiVersions[apiKey]; ok {
+		supported := proto.SupportedByDriver[apiKey]
+		if min(supported.MaxVersion, requested.MaxVersion) >= max(supported.MinVersion, requested.MinVersion) {
+			return min(supported.MaxVersion, requested.MaxVersion)
+		}
+		return 0
```

Comment:
remove it?

Reply:
Why ?  I am not sure.  Properly done this function should return  *int  or (int, error) because the situation where supported by server set and supported by client set make an empty set when intersected.  But for simplification I just downgrade protocol to 0 in such cases.

Reply:
I mean the code anyway return 0 on line 107, or I missed something?

Reply:
Nope, you are right, sorry. Fixed